src/query/epa/epa3.rs

```diff
@@ -284,6 +284,14 @@ impl EPA {
                 let dist4 = self.faces[3].normal.dot(&self.vertices[3].point.coords);
                 self.heap.push(FaceId::new(3, -dist4)?);
             }
+
+            if !(proj_inside1 || proj_inside2 || proj_inside3 || proj_inside4) {
+                // Related issues:
+                // https://github.com/dimforge/parry/issues/253
+                // https://github.com/dimforge/parry/issues/246
+                log::debug!("Hit unexpected state in EPA: proj_inside1, proj_inside2, proj_inside3 and proj_inside4 are all false. At least one of them should be true.");
```

Comment:
```suggestion
                log::debug!("Hit unexpected state in EPA: failed to project the origin on the initial simplex.");
```

Reply:
I just realized something. Does the projection really "fail"? In my test, all of the `Face::new` calls report `OnEdge`  location for the projection instead of `OnFace`. But could we not also consider `OnVertex` and `OnEdge` workable locations for the projections?

Reply:
```rust
);
        let (_, loc) = tri.project_local_point_and_get_location(&Point::<Real>::origin(), true);

        match loc {
            TrianglePointLocation::OnFace(_, bcoords) => {
                (Self::new_with_proj(vertices, bcoords, pts, adj), true)
            }
            _ => (Self::new_with_proj(vertices, [0.0; 3], pts, adj), false),
        }
```

Was there an assumption, that if the location is `OnEdge` or `OnVertex`, origin was not touching the triangle in the first place? In this case the origin seems to be touching an edge already, so maybe instead of relying on `OnFace`, we should check whether the projection is sufficiently close to the origin? Two of the projections are so close to origin, that the distance could be attributed to numerical instability:

```
[crates/parry3d/../../src/query/epa/epa3.rs:101:9] &proj = PointProjection {
    is_inside: true,
    point: [
        0.0,
        0.0,
        -1.1920929e-7,
    ],
}
[crates/parry3d/../../src/query/epa/epa3.rs:101:9] &proj = PointProjection {
    is_inside: true,
    point: [
        0.0,
        0.0,
        -1.1920929e-7,
    ],
}
```